strawberryfields/engine.py

```diff
@@ -420,7 +420,10 @@ def run(self, program, *, args=None, compile_options=None, **kwargs):
 
         if not isinstance(program, collections.abc.Sequence) and program.type == "tdm":
             # At this time we do not support lists of tdm programs
-            program.unroll()
+            shots = kwargs.get("shots", 1)
+            program.unroll(shots=shots)
+            # set shots to 1 so that the gaussian backend works
```

Comment:
Great that you added this comment!

Reply:
Good catch! I have to admit, even with this comment I was still confused for a second 😆 I suggest adding a few more details:

```suggestion
            # Shots >1 for a TDM program simply corresponds to creating
            # multiple copies of the program, and appending them to run sequentially.
            # As a result, we set the backend shots to 1 for the Gaussian backend.
```